cpp/src/transform/row_bit_count.cu

```diff
@@ -398,26 +401,29 @@ __device__ size_type row_size_functor::operator()<struct_view>(column_device_vie
  * @param cols An span of column_device_views representing a column hierarchy
  * @param info An span of column_info structs corresponding the elements in `cols`
  * @param output Output span of size (# rows) where per-row bit sizes are stored
+ * @param segment_length The number of rows in each segment for which the total size is computed
  * @param max_branch_depth Maximum depth of the span stack needed per-thread
  */
 CUDF_KERNEL void compute_row_sizes(device_span<column_device_view const> cols,
                                    device_span<column_info const> info,
                                    device_span<size_type> output,
+                                   size_type segment_length,
```

Comment:
Probably should rename this to `compute_segment_sizes` as well, and add a comment in the description that a segment size of 1 yields the old per-row behavior.